cmd/queue/main.go

```diff
@@ -192,6 +192,11 @@ func main() {
 		flush(logger)
 		os.Exit(1)
 	case <-ctx.Done():
+		// If container-freezer enabled, resume container so that normal shutdown can occur
```

Comment:
I think we should avoid leaking the idea of container-freezer in here (and in general think about how to avoid leaking it in to QP). The concurrency state endpoint is supposed to be generic, so I'd say we should try to keep things generic and talk about what's happening rather than imperatively telling it to resume. WDYT about having an explicit "shutdown" hook when we receive a term signal? (that also has the advantage that an implementation might want to act differently on shutdown - for example keeping the container unpaused until it exists to allow for background tasks, even if a further Pause occurs)

Separately Im thinking about whether we should change "Pause" and "Resume" to "Active" and "Inactive" (i.e. tell the endpoint the state, rather than requesting an imperative action) or something like that (and if so "Shutdown" => "Terminating").